virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -332,12 +335,38 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
     };
   }
 
+  private generateSessionEventsProvider(): Angular2PluginSessionEvents {
+    let loginSubject = new Subject<void>();
+    let logoutSubject = new Subject<void>();
+    let sessionExpireSubject = new Subject<void>();
+    this.authenticationManager.loginScreenVisibilityChanged.subscribe((eventReason: LoginScreenChangeReason) => {
```

Comment:
You should not have the subscribe here. Rather, this subscribe should be part of the window manager service constructor, and you store each of the subjects in a map of key=windowid, and value=the 3 subjects, then during a event that comes from the visibility change, you can run through the list and call the event on the subjects that are still valid.
However, upon closewindow or destroywindow ( i forgot which) you can clean up the map so that things dont get called.